onnxruntime/core/optimizer/mobile_transformer.cc

```diff
@@ -0,0 +1,240 @@
+    if (!act_node) {
+    } else if (graph_utils::IsSupportedOptypeVersionAndDomain(*act_node, "Relu", {6, 13, 14}) ||
+                     graph_utils::IsSupportedOptypeVersionAndDomain(*act_node, "Sigmoid", {6, 13}) ||
+                     graph_utils::IsSupportedOptypeVersionAndDomain(*act_node, "Tanh", {6, 13}) ||
+                     graph_utils::IsSupportedOptypeVersionAndDomain(*act_node, "LeakyRelu", {6})) {
+    } else if (graph_utils::IsSupportedOptypeVersionAndDomain(*act_node, "Clip", {6, 11, 12, 13})) {
+      float min, max;
+      if (!optimizer_utils::GetClipConstantMinMax(graph_viewer.GetGraph(), *act_node, min, max)) {
+        act_node = nullptr;
+      }  
+    } else {
+      act_node = nullptr;
+    }
+
+
+    NodesToOptimizeIndicesBuilder builder{};
+    builder.target_node = conv_node->Index();
+    builder.output_nodes = {add_node->Index()};
+    if (act_node) {
+      builder.output_nodes.push_back(act_node->Index());
+    }
+    return builder.Build();
+  }
+
+  const Node* MultiSelectBinary(const Node& node, bool add_node) const {
+    InlinedVector<Node*> inputs_node;
+    if (!add_node) {
+      // FIX ME, not sure if it deserve to implement multiplication-relu fusion
+      return nullptr;
+    }
+    constexpr int32_t kTensorDims = 4;
+    const auto& input_defs = node.InputDefs();
+
+
+    for (auto producer_node_ptr = node.InputNodesBegin(); producer_node_ptr != node.InputNodesEnd(); ++producer_node_ptr) {
+      Node* producer_node = const_cast<Node*>(&(*producer_node_ptr));
+      inputs_node.push_back(producer_node);
+    }
+    if (inputs_node.size() == 0) {
+      return nullptr;
+    }
+    size_t input_defs_count = input_defs.size();
+
+    // Test if all of inputs have an equal shape.
+    bool all_shapes_match = true;
+    auto* input_0_shape = input_defs[0]->Shape();
+    for (size_t n = 1; n < input_defs_count; n++) {
+      for (int i = 0; i < kTensorDims; i++) {
+        // Check if ONNX shape inferencing has computed a precise dimension value.
+        auto* input_n_shape = input_defs[n]->Shape();
+        if ((input_0_shape == nullptr) || (input_n_shape == nullptr)) {
+          all_shapes_match = false;
+        } else {
+          auto& input_0_dim = input_0_shape->dim(i);
+          auto& input_n_dim = input_n_shape->dim(i);
+          if (!utils::HasDimValue(input_0_dim) ||
+              !utils::HasDimValue(input_n_dim) ||
+              (input_0_dim.dim_value() <= 0) ||
+              (input_0_dim.dim_value() != input_n_dim.dim_value())) {
+            if (!utils::HasDimParam(input_0_dim) ||
+                !utils::HasDimParam(input_n_dim) ||
+                (input_0_dim.dim_param() != input_n_dim.dim_param())) {
+              all_shapes_match = false;
+              break;
+            }
+          }
+        }
+      }
+    }
+    if (all_shapes_match) {
+      // If one of the inputs to the Add/Sum node is a convolution, then
```

Comment:
Isn't one of the inputs always a Conv given the selection code is triggered on a Conv node and this is a consumer node of the output from that?